### PR TITLE
Lucene: Make  the operation under permission more meaningful and consistent

### PR DESCRIPTION
**Pull request by infofromca**

### CODE REVIEW

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs [hunk 1]
```diff
@@ -82,7 +82,7 @@ public async Task<IActionResult> Index(string id, string q, PagerParameters page
                 return View(new SearchIndexViewModel
                 {
                     Pager = pager,
-                    IndexName = id,
+                    IndexName = indexName,
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs]: indexName is not declared anywhere

Comment (infofromca) [reply src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs]: ln 59  var indexName = "Search";

File: src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs [hunk 2]
```diff
@@ -50,7 +50,7 @@ public async Task<IActionResult> Index(string id, string q, PagerParameters page
         {
             if (!await _authorizationService.AuthorizeAsync(User, Permissions.QueryLuceneSearch))
             {
-                return NotFound();
+                return Unauthorized();
```
Comment (Skrypt) [inline src/OrchardCore.Modules/OrchardCore.Lucene/Controllers/SearchController.cs]: Makes sense.

**Comment by Skrypt** [comment]
LGTM

**Pull request merged**